### PR TITLE
feat(workflow): Including GroupOwner in Group serializer

### PR DESCRIPTION
**Pull request by iProgramStuff**
Adding support for a new `expand` attribute in the `StreamGroupSerializerSnuba` serializer: `owners`.

Using `expand=owners` will now attach GroupOwner information as a list of dicts. Each dict will contain `type`, `owner`, and `date_added`.

`type` will be a string from `GROUP_OWNER_TYPE`
`owner` will be a string of format `owner_type:id`, e.g. `team:12`, `user:41`

### CODE REVIEW

File: src/sentry/api/endpoints/organization_group_index.py [hunk 1]
```diff
@@ -115,6 +115,9 @@ def get(self, request, organization):
         expand = request.GET.getlist("expand", [])
         collapse = request.GET.getlist("collapse", [])
         has_inbox = features.has("organizations:inbox", organization, actor=request.user)
+        has_workflow_owners = features.has(
+            "organizations:workflow-owners", organization, actor=request.user
```
Comment (davidenwang) [inline src/sentry/api/endpoints/organization_group_index.py:119]: do we need to add this to the features `default_manager`?

Comment (iProgramStuff) [reply src/sentry/api/endpoints/organization_group_index.py:119]: Yeah :)
https://github.com/getsentry/sentry/pull/22241/

Comment (davidenwang) [reply src/sentry/api/endpoints/organization_group_index.py:119]: oh my b didn't see that

Comment (iProgramStuff) [reply src/sentry/api/endpoints/organization_group_index.py:119]: I added it after your comment 😎

File: src/sentry/models/groupowner.py [hunk 2]
```diff
@@ -57,3 +57,18 @@ def owner(self):
...
 
         return Actor.from_actor_identifier(self.owner_id())
+
+
+def get_owner_details(group_list):
+    group_ids = [g.id for g in group_list]
+    group_owners = GroupOwner.objects.filter(group__in=group_ids)
+    owner_details = {
+        go.group_id: {
+            "type": go.type,
```
Comment (markstory) [inline src/sentry/models/groupowner.py:67]: Should we convert the type back to the string name? We often convert integer enums back to string values like in transaction.status, activity types, group subscription reasons and more. The string values are simpler for end users as they don't need to refer to an enum to know what the current value is.

File: tests/snuba/api/endpoints/test_organization_group_index.py [hunk 3]
```diff
@@ -739,21 +741,8 @@ def test_inbox_fields(self):
...
-                project_id=self.project.id,
-            )
             add_group_to_inbox(event.group, GroupInboxReason.NEW)
-
-            query = u"server:example.com"
-            query += u" status:unresolved"
-            query += u" active_at:" + iso_format(before_now(seconds=350))
-            query += u" first_seen:" + iso_format(before_now(seconds=500))
-
+            query = u" status:unresolved"
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_group_index.py:745]:
```suggestion
            query = u"status:unresolved"
```

File: src/sentry/api/endpoints/organization_group_index.py (hunk 1 above)
Comment (davidenwang) [reply src/sentry/api/endpoints/organization_group_index.py:119]: oh haha i take it back then 😎

File: src/sentry/models/groupowner.py (hunk 2 above)
Comment (iProgramStuff) [reply src/sentry/models/groupowner.py:67]: Will do, thanks!

File: tests/snuba/api/endpoints/test_organization_group_index.py [hunk 4]
```diff
@@ -739,21 +742,8 @@ def test_inbox_fields(self):
                 data={"timestamp": iso_format(before_now(seconds=500)), "fingerprint": ["group-1"]},
                 project_id=self.project.id,
             )
-            self.store_event(
```
Comment (davidenwang) [inline tests/snuba/api/endpoints/test_organization_group_index.py:742]: just curious cuz i don't touch the backend as often, but were these not needed?

**Review by davidenwang — APPROVED** [review]
lgtm

File: tests/snuba/api/endpoints/test_organization_group_index.py (hunk 4 above)
Comment (iProgramStuff) [reply tests/snuba/api/endpoints/test_organization_group_index.py:742]: Yeah they were totally irrelevant to what was actually being tested and seemed fine to be removed (...aka I don't know why I put them there in the first place 😄 )

File: src/sentry/models/groupowner.py [hunk 5]
```diff
@@ -15,6 +15,12 @@ class GroupOwnerType(Enum):
     OWNERSHIP_RULE = 1
 
 
+GROUP_OWNER_TYPE = {
+    GroupOwnerType.SUSPECT_COMMIT: "Suspect Commit",
+    GroupOwnerType.OWNERSHIP_RULE: "Ownership Rule",
```
Comment (iProgramStuff) [inline src/sentry/models/groupowner.py:20]: So I'm sending "Suspect Commit"/"Ownership Rule". Not sure if the frontend would rather something like "suspect_commit" / "ownership_rule"?

Comment (davidenwang) [reply src/sentry/models/groupowner.py:20]: should be fine on the frontend either way, not sure if there is a standard tho for snakecase :/

Comment (markstory) [reply src/sentry/models/groupowner.py:20]: Many of the enums we have in the UI today use `camelBack` form. So `suspectCommit` and `ownershipRule` would work.

Comment (iProgramStuff) [reply src/sentry/models/groupowner.py:20]: Changing to that 👍

**Comment by wedamija** [comment]
@iProgramStuff Since this returns a list of objects, does it make more sense for this to be a separate endpoint? `groups/1/owners` or something along those lines?

**Pull request merged**